pkg/apis/agones/v1/gameserver.go

```diff
@@ -533,21 +560,21 @@ func (gs *GameServer) Pod(sidecars ...corev1.Container) (*corev1.Pod, error) {
 
 	gs.podObjectMeta(pod)
 
-	i, gsContainer, err := gs.FindGameServerContainer()
-	// this shouldn't happen, but if it does.
-	if err != nil {
-		return pod, err
-	}
-
 	for _, p := range gs.Spec.Ports {
 		cp := corev1.ContainerPort{
 			ContainerPort: p.ContainerPort,
 			HostPort:      p.HostPort,
 			Protocol:      p.Protocol,
 		}
-		gsContainer.Ports = append(gsContainer.Ports, cp)
+		err := gs.ApplyToPodContainer(pod, *p.Container, func(c corev1.Container) corev1.Container {
```

Comment:
This should also be wrapped in a FeatureContainerPortAllocation - such that the previous logic stays in place for when it's disabled, and the new logic is in place for the new feature.

Then we'll need to split out a test for the enabled and disabled unit tests. ([Here's an example](https://github.com/googleforgames/agones/pull/1399/files#diff-a2a0019b8308830fa9857437be51f29dR910-R914))

Reply:
I left the old test in place, and enabled the flag for the new test. They should both pass now 🤞